### ISSUE TITLE
2024 HA can't load ims platform

### ISSUE THREAD

**Issue by Gugulator** [GuyKh/ims-custom-component#64]
Logger: homeassistant.setup
Source: setup.py:403
First occurred: 00:42:38 (1 occurrences)
Last logged: 00:42:38

Unable to prepare setup for platform 'ims.sensor': Platform not found (Exception importing custom_components.ims.sensor).

**Comment by Gugulator** [comment]
Logger: homeassistant.loader
Source: loader.py:842
First occurred: 00:42:38 (1 occurrences)
Last logged: 00:42:38

Unexpected exception importing platform custom_components.ims.sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 842, in get_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 859, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/ims/sensor.py", line 325, in <module>
    class ImsSensor(ImsEntity, SensorEntity, ImsSensorEntityDescription):
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases

**Comment by vadimnazaroff** [comment]
Having the same issue:
Logger: homeassistant.loader
Source: loader.py:842
First occurred: 11:21:34 (1 occurrences)
Last logged: 11:21:34

Unexpected exception importing platform custom_components.ims.sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 842, in get_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 859, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/ims/sensor.py", line 325, in <module>
    class ImsSensor(ImsEntity, SensorEntity, ImsSensorEntityDescription):
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases

**Comment by MarkLuk** [comment]
Same issue 
```
Logger: homeassistant.loader
Source: loader.py:842
First occurred: 8:26:24 AM (1 occurrences)
Last logged: 8:26:24 AM

Unexpected exception importing platform custom_components.ims.sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 842, in get_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 859, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/ims/sensor.py", line 325, in <module>
    class ImsSensor(ImsEntity, SensorEntity, ImsSensorEntityDescription):
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

**Comment by mulikg** [comment]
This entity is no longer being provided by the ims integration. If the entity is no longer in use, delete it in settings

**Comment by vadimnazaroff** [comment]
is there any ETA for issue fix?

**Comment by GuyKh** [comment]
The issue is because of EntityDescription changes introduced in 2024.1. I'm abroad and on vacation at the moment, so it'll take a while.

**Comment by vadimnazaroff** [comment]
@GuyKh Sure thing. Enjoy it!

**Comment by sharonoren1** [comment]
Same issue for me.
I'll wait for a solution

**Comment by refaelsommer** [comment]
I see there is a PR fixing this issue from @GuyKh, Who is in-charge of merging it so we can use it?

**Comment by GuyKh** [comment]
I am. It's not ready yet

**Comment by Gugulator** [comment]
fixed in new IMS version 0.1.20. thanks

**Comment by vadimnazaroff** [comment]
fixed in 0.1.20. Thanks @GuyKh !
